app/components/special-coverage/special-coverage-edit/special-coverage-edit-directive.js

```diff
@@ -66,8 +66,47 @@ angular.module('specialCoverage.edit.directive', [
           return promise;
         };
 
+        // Maps ID to Object for formatting (since only tunic_campaign_id stored on SpecialCoverage object)
+        $scope.tunicCampaignIdMapping = {};
+        $scope.campaignsRetrieved = function (response) {
+          response.data.results.forEach(function (result) {
+            $scope.tunicCampaignIdMapping[result.id] = result;
+          });
+          var tunicCampaignIds = [];
+          response.data.results.forEach(function (result) {
+            tunicCampaignIds.push(result.id);
+          });
+          return tunicCampaignIds;
+        };
+
+        $scope.getCampaign = function (tunicCampaignId) {
+          return $http.get('http://tunic.local/api/v1/campaign/' + tunicCampaignId + '/', {
+            headers: {
+              'Authorization': 'Token 246bce7a5ddaed1fd497ed9b53d0a2281e3928f5'
+            }
+          }).then(function (result) {
```

Comment:
As trivial as it is, we should move this anonymous function into a named one to make it easier to unit test